python/ray/serve/deployment.py

```diff
@@ -125,7 +126,10 @@ def _validate_name(self, name: str):
 
         # name does not contain #
         if "#" in name:
-            raise ValueError("name cannot contain #")
+            warnings.warn(
+                f"Name cannot contain '#' character, it will be removed in future releases. "
+                f"Current name: {name}."
```

Comment:
```suggestion
                f"Deployment names cannot contain '#' character, this will raise an error in a future release. "
                f"Current name: {name}."
```

Reply:
just making it more precise (users might not know what "name" this is referring to)

Reply:
done